pymoab/range.pyx

```diff
@@ -35,4 +35,7 @@ cdef class Range(object):
 
     def __getitem__(self, int index):
         cdef moab.EntityHandle rtn = deref(self.inst)[index]
-        return rtn
+        if index < self.size():
+            return rtn
+        else:
+            raise StopIteration
```

Comment:
@scopatz can do. Thanks!

I'm not very familiar with how iterables are supposed to work. I'm assuming that if the object doesn't have an __iter__ method then it will try to use __getitem__ method to do the same thing?